pkg/yang/entry.go

```diff
@@ -1024,11 +1026,30 @@ func addExtraKeywordsToLeafEntry(n Node, e *Entry) {
 			"reference",
 			"status",
 			"when":
-			e.Extra[name] = append(e.Extra[name], fv.Interface())
+			if !fv.IsNil() {
+				addToExtrasSlice(fv, name, e)
+			}
 		}
 	}
 }
 
+func addToExtrasSlice(fv reflect.Value, name string, e *Entry) {
```

Comment:
I wonder why this processing is necessary?

Reply:
Mainly this is non-backwards-compatible. I'm wondering if this can be done a layer up?

Reply:
Thanks for reviewing @wenovus 

`Extra` is defined as a `map[string][]interface{}` and the `[]interface{}` had only 1 item ever added to it. Regardless of whether the item to be added was a single struct or an array of structs it was being added as 1 entry in the `[]interface{}`.

This created an unnecessary 2 step situation at time of decoding, where the item added was an array:
1) Open up the `[]interface{}` and discover it has only 1 entry
2) open up that 1 entry and iterate through its entries

What I am doing here, is when the item to be added is an array, is to add the entries individually to the `Extra` as I presumed it was meant to be used.

An alternative way to simplify it at a higher level would be to declare the `Extra` variable instead as `map[string]interface{}` and in this way the item could be set on it directly, regardless of whether single or array of struct.

I did not do this though as I thought changig the defintion of `Extra` might break existing dependencies.